### PR TITLE
fcppt: 3.2.2 -> 3.5.0

### PR DESCRIPTION
**Pull request by pmiddend**
###### Motivation for this change

Regular update. Note the added `metal` dependency. This package isn't in nixpkgs yet. I wasn't sure if I should include it "officially", so it's just an embedded derivation here. Is that okay?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/fcppt/default.nix [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
+      owner = "brunocodutra";
+      repo = "metal";
+      rev = "v${version}";
+      sha256 = "07n1aqyaixbd66l24km5ip3pkmidkx9m3saygf7cfp6vvbgmi42l";
+    };
+
+    nativeBuildInputs = [ cmake ];
+    buildInputs = [];
+
+    cmakeFlags = [];
```
Comment (bhipple) [inline pkgs/development/libraries/fcppt/default.nix:17]:
```suggestion
```
Same for `buildInputs`; if you don't have any, let's just omit it.

**Review by bhipple — CHANGES_REQUESTED** [review]
I'd say it's better to do a `metal: init at 2.1.1` commit and depend on the pkg cleanly from the distribution, rather than inlining it, unless you're sure that this lib dependency will never be used outside of this one derivation (e.g., it's a private lib).  

Also, as a nitpick, we're slowly going through and quoting all of the URLs in fields like `meta.homepage` for https://github.com/NixOS/rfcs/pull/45

**Comment by pmiddend** [comment]
@bhipple Thanks for the review! I've fixed the things you mentioned.

**Comment by bhipple** [comment]
LGTM!

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 84880` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
<br>- fcppt
<br>- metal
</details>

**Review by bhipple — APPROVED** [review]

**Comment by pmiddend** [comment]
Thanks!

**Pull request merged**